File: BondPrototype/Models/ApiModels/MovieResult.cs
namespace BondPrototype.Models;

public class MovieResult
{
    public int Id { get; set; }
    public string Title { get; set; }
    public byte Rating { get; set; }
    public DateTime ReleaseDate { get; set; }
    public string MoviePoster { get; set; }
    public string Description { get; set; }

    public record struct Director(int Id, string Name, DateTime? DateOfBirth);

    public Director DirectedBy { get; set; }

    public record struct Actor(int Id, string Name, DateTime? DateOfBirth);

    public List<Actor> Actors { get; set; }

    public record struct Award(AwardType Type, string Name);

    public List<Award> Awards { get; set; }
}

File: BondPrototype/Angular/src/app/api/models/api-models/movie-result.ts
import {AwardType} from '../award';


export class MovieResult {
    public id: number;
    public title: string;
    public rating: number;
    public releaseDate: Date;
    public moviePoster: string;
    public description: string;
    public get savo() { return `Hello there..`; }
    public directedBy: MovieResult.Director;
    public actors: MovieResult.Actor[];
    public awards: MovieResult.Award[];
}

export namespace MovieResult {
    export class Director {
        constructor(public id: number, public name: string, public dateOfBirth: Date | null) {
        }
    }
    export class Actor {
        constructor(public id: number, public name: string, public dateOfBirth: Date | null) {
        }
    }
    export class Award {
        constructor(public type: AwardType, public name: string) {
        }
    }
}
